concat-kdf/tests/tests.rs

```diff
@@ -0,0 +1,200 @@
+use digest::{Digest, FixedOutputReset};
+use hex_literal::hex;
+use sha2::{Sha224, Sha256, Sha512};
+
+struct Test<'a> {
+    secret: &'a [u8],
+    other_info: &'a [u8],
+    expected_key: &'a [u8],
+}
+
+fn test_input_output<D>(tests: &[Test])
+where
+    D: Digest + FixedOutputReset,
+{
+    for Test {
+        secret,
+        other_info,
+        expected_key,
+    } in tests.iter()
+    {
+        for key_length in 1..expected_key.len() {
+            let mut key = vec![0u8; key_length];
+            assert!(concat_kdf::derive_key_into::<D>(secret, other_info, &mut key).is_ok());
```

Comment:
maybe create custom assert/macro with a speaking name e.g. `assert_key_derivation(...)`
which does all the needed checks (the 2 asserts you are currently using)